third-party/realdds/src/dds-device-broadcaster.cpp

```diff
@@ -270,13 +270,27 @@ bool dds_device_broadcaster::send_device_info_msg( const dds_device_handle & han
     return false;
 }
 
+template< class Array >
+void copy_to_array( std::string const & source, Array & dest )
+{
+    // Workaround for "warning C4996: 'strcpy': This function or variable may be unsafe"
+    // Same with strncpy.
+    //
+    assert( dest.size() > 1 );
+    size_t const cch_copied = source.copy( dest.data(), dest.size() - 1 );
+    dest.at( cch_copied ) = 0;  // copy() does not append a null character
+    // Note that this does not pad the rest of the array with nulls is source was shorter!
+    //strncpy( dest.data(), source.c_str(), dest.size() );
+    //dest.back() = 0;  // make sure it's 0-terminated
+}
+
 void dds_device_broadcaster::fill_device_msg( const device_info & dev_info,
                                               librealsense::dds::topics::raw::device_info & msg ) const
 {
-    strcpy( msg.name().data(), dev_info.name.c_str() );
```

Comment:
Not sure if `std::copy` is an option here but why not just replacing with `strcpy_s` usage?
Won't it remove the warning?

Reply:
Because of linux. It likely won't have it.

Reply:
👍